src/frontend/js/components/LanguageSelector/index.spec.tsx

```diff
@@ -0,0 +1,45 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { LanguageSelector } from '.';
+import { IntlProvider } from 'react-intl';
+
+describe('<LanguageSelector />', () => {
+  it('shows a dropdown menu that allows the user to change languages', () => {
+    const languages = {
+      en: {
+        code: 'en',
+        name: 'Anglais',
+        url: '/switch/to/en/',
+      },
+      fr: {
+        code: 'fr',
+        name: 'Français',
+        url: '/switch/to/fr/',
+      },
+    };
+
+    render(
+      <IntlProvider locale="en">
+        <LanguageSelector currentLanguage="fr" languages={languages} />
+      </IntlProvider>,
+    );
+
+    const button = screen.getByRole('button', {
+      name: 'Select a language: Français',
+    });
+
+    userEvent.click(button);
+
+    screen.getByRole('listbox', { name: 'Select a language:' });
+    screen.getByRole('option', { name: 'Switch to Anglais' });
+    screen.getByRole('link', { name: 'Switch to Anglais' });
+    screen.getByRole('option', {
+      name: 'Switch to Français (currently selected)',
```

Comment:
this is shown on hover and is not being translated

Reply:
It's expected not to be translated as we have not done the translations for this PR yet.

As for being shown on hover, this is a consequence of being the title of the link. Is it an issue? I think we're supposed to put this sort of thing as part of a link title when the link is "doing something", not just a regular link

Reply:
Ah ok. Showing on hover is great. I was mentioning the absence of a translation.
You're right, we'll provide translations in the coming release.